test/e2e/kubectl.go

```diff
@@ -1695,6 +1695,83 @@ metadata:
 			}
 		})
 	})
+
+	framework.KubeDescribe("Kubectl deployment reaper", func() {
+		var nsFlag string
+		var name string
+		var labelFlag string
+
+		BeforeEach(func() {
+			nsFlag = fmt.Sprintf("--namespace=%v", ns)
+			name = "e2e-test-nginx-deployment"
+			labelFlag = fmt.Sprintf("-l run=%v", name)
+		})
+
+		// No conformance label added as it is an explicit version skew test
+		It("should delete all replicasets of cluster", func() {
+			// Create deployment with a specific label so that replicasets and pods also have the label
+			By("running the image " + nginxImage)
+			framework.RunKubectlOrDie("run", name, "--image="+nginxImage, nsFlag, labelFlag)
+
+			// If the pod is created successfully, it means deployment and replicasets are created successfully too
+			By("verifying the pod controlled by " + name + " gets created")
+			label := labels.SelectorFromSet(labels.Set(map[string]string{"run": name}))
+			err := testutils.WaitForPodsWithLabelRunning(c, ns, label)
+			if err != nil {
+				framework.Failf("Failed getting pod controlled by %s: %v", name, err)
+			}
+
+			By("deleting the deployment " + name)
+			framework.RunKubectlOrDie("delete", "deployment", name, nsFlag)
+
+			var pollErr error
+
+			// Verify deletion completion in this order: deployment, replicasets, pods
+			// In accord to how garbage processor works
+
+			By("verifying deployment was deleted")
+			wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+				deploymentList, err := c.Extensions().Deployments(ns).List(metav1.ListOptions{LabelSelector: label.String()})
+				if err != nil || len(deploymentList.Items) != 0 {
+					pollErr = err
+					return false, nil
+				}
+				pollErr = nil
```

Comment:
Take a look at `wait.PollImmediate` godoc. You should get the returned error from `wait.PollImmediate`, and inside this function return true when the deployment is not found (the condition you're waiting for), and return false otherwise. Then you check the returned error from `wait.PollImmediate`.